cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -791,6 +854,22 @@ def configure_master_services():
     api_opts.add('insecure-bind-address', '127.0.0.1')
     api_opts.add('insecure-port', '8080')
     api_opts.add('storage-backend', 'etcd2')  # FIXME: add etcd3 support
+
+    # TODO: investigate if it's possible to use config file to store args
+    # Handle api-extra-args config option
+    to_add, to_remove = get_config_args()
+    for arg in to_add:
+        hookenv.log('Adding option: {} {}'.format(arg[0], arg[1]))
+        # Make sure old value is gone
+        api_opts.destroy(arg[0])
+        api_opts.add(arg[0], arg[1])
+    for arg in to_remove:
+        hookenv.log('Removing option: {}'.format(arg))
+        api_opts.destroy(arg)
+        # We need to "unset" options by settig their value to "null" string
+        cmd = ['snap', 'set', 'kube-apiserver', '{}=null'.format(arg)]
+        check_call(cmd)
```

Comment:
I think this could have confusing behavior when removing overridden args. What happens if I do this?

```
juju config kubernetes-master api-extra-args="storage-backend=etcd3"
juju config kubernetes-master api-extra-args=""
```

1. By default, the charm uses `storage-backend=etcd2`.
2. Setting api-extra-args will allow us to override it with `storage-backend=etcd3`, which seems reasonable.
3. Once we clear api-extra-args, it will set it to `storage-backend=null`, instead of reverting to etcd2.

I think it'd be worthwhile to fix the logic here so it reverts back properly. That or prevent overrides from happening if that's something we don't want.